### PR TITLE
Add exception control when there are no records found

### PR DESCRIPTION
**Pull request by ruaraya**
Add exception control when there are no records found

### CODE REVIEW

File: src/Spd.Resource.Repository/Licence/LicenceRepository.cs [hunk 1]
```diff
@@ -17,15 +18,23 @@ internal class LicenceRepository : ILicenceRepository
...
+                .Expand(i => i.spd_LicenceHolder_contact)
+                .Expand(i => i.spd_LicenceHolder_account)
+                .Expand(i => i.spd_CaseId)
+                .Where(l => l.spd_licenceid == licenceId)
+                .FirstOrDefaultAsync(ct);
+        }
+        catch (DataServiceQueryException ex)
+        {
+            if (ex.Response.StatusCode == 404)
+                throw new ArgumentException($"cannot find the licence with licence Id : {licenceId}");
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/Licence/LicenceRepository.cs:34]: No, Ruben, what we want is when there is no record found, we return null.

File: src/Spd.Resource.Repository/LicenceApplication/LicenceApplicationRepository.cs [hunk 2]
```diff
@@ -136,14 +137,24 @@ public async Task<LicenceApplicationCmdResp> SaveLicenceApplicationAsync(SaveLic
...
+                .Expand(a => a.spd_ApplicantId_contact)
+                .Expand(a => a.spd_application_spd_licencecategory)
+                .Expand(a => a.spd_CurrentExpiredLicenceId)
+                .Where(a => a.spd_applicationid == licenceApplicationId)
+                .SingleOrDefaultAsync(ct);
+        }
+        catch (DataServiceQueryException ex)
+        {
+            if (ex.Response.StatusCode == 404)
+                throw new ArgumentException($"cannot find application with Id : {licenceApplicationId}");
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/LicenceApplication/LicenceApplicationRepository.cs:153]: the same as above.

File: src/Spd.Resource.Repository/Licence/LicenceRepository.cs (hunk 1 above)
Comment (ruaraya) [reply src/Spd.Resource.Repository/Licence/LicenceRepository.cs]: Fixed

File: src/Spd.Resource.Repository/LicenceApplication/LicenceApplicationRepository.cs (hunk 2 above)
Comment (ruaraya) [reply src/Spd.Resource.Repository/LicenceApplication/LicenceApplicationRepository.cs]: Fixed

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**